kedro/framework/cli/catalog.py

```diff
@@ -50,10 +51,15 @@ def list_datasets(metadata: ProjectMetadata, pipeline, env):
     title = "Datasets in '{}' pipeline"
     not_mentioned = "Datasets not mentioned in pipeline"
     mentioned = "Datasets mentioned in pipeline"
+    factories = "Datasets generated from factories"
 
     session = _create_session(metadata.package_name, env=env)
     context = session.load_context()
-    datasets_meta = context.catalog._data_sets  # noqa: protected-access
+
+    data_catalog = context.catalog
+    datasets_meta = data_catalog._data_sets
+    catalog_ds = set(data_catalog.list())
+    datasets_meta = context.catalog._data_sets
```

Comment:
In line 60 `datasets_meta` is defined as `data_catalog._data_sets` but then in line 62 it's immediately overwritten as `datasets_meta = context.catalog._data_sets`, is that on purpose? And what exactly do you mean with "meta" here?

Reply:
Thanks for catching this! I suspect this was introduced by a poor resolve on the merge conflicts, I've removed the redundant lines. 

The variable `datasets_meta` predates the changes in this PR, from my understanding using meta refers to the protected dictionary `catalog._data_sets` as opposed to the public _FrozenDataSets object `catalog.datasets`

Reply:
Ah! Thanks for clarifying!